app/src/main/java/at/bitfire/icsdroid/db/dao/SubscriptionsDao.kt

```diff
@@ -0,0 +1,132 @@
+package at.bitfire.icsdroid.db.dao
+
+import android.content.Context
+import android.database.SQLException
+import androidx.annotation.WorkerThread
+import androidx.lifecycle.LiveData
+import androidx.room.*
+import at.bitfire.icsdroid.db.AppDatabase
+import at.bitfire.icsdroid.db.entity.Subscription
+
+/**
+ * Creates an interface with all the subscriptions made in the database.
+ * @author Arnau Mora
+ * @see AppDatabase
+ */
+@Dao
+interface SubscriptionsDao {
+    companion object {
+        /**
+         * Alias for [AppDatabase.getInstance] -> [AppDatabase.subscriptionsDao].
+         * @param context The context that is requesting access to the dao.
+         */
+        fun getInstance(context: Context) = AppDatabase.getInstance(context).subscriptionsDao()
+    }
+
+    /**
+     * Adds one or more new subscriptions to the database.
+     *
+     * **This doesn't add the subscription to the system's calendar.** It's preferable to use
+     * [Subscription.create] adding new subscriptions.
+     * @author Arnau Mora
+     * @param subscriptions All the subscriptions to be added.
+     * @throws SQLException If any error occurs with the request.
+     */
+    @Insert
+    @WorkerThread
+    fun add(vararg subscriptions: Subscription)
+
+    /**
+     * Gets a [LiveData] with all the made subscriptions. Updates automatically when new ones are added.
+     * @author Arnau Mora
+     * @throws SQLException If any error occurs with the request.
+     */
+    @Query("SELECT * FROM subscriptions")
+    fun getAllLive(): LiveData<List<Subscription>>
+
+    /**
+     * Gets a list of all the made subscriptions.
+     * @author Arnau Mora
+     * @throws SQLException If any error occurs with the request.
+     */
+    @WorkerThread
+    @Query("SELECT * FROM subscriptions")
+    fun getAll(): List<Subscription>
+
+    /**
+     * Gets an specific [Subscription] by its id ([Subscription.id]).
+     * @author Arnau Mora
+     * @param id The id of the subscription to fetch.
+     * @return The [Subscription] indicated, or null if any.
+     * @throws SQLException If any error occurs with the request.
+     */
+    @WorkerThread
+    @Query("SELECT * FROM subscriptions WHERE id=:id LIMIT 1")
```

Comment:
Do we need LIMIT 1? The `id` is a primary key so there must not be two entries with the same ID.

Reply:
eb9418adad0d12e701b74a1de0f2aea7f63d57b5